include/envoy/access_log/access_log.h

```diff
@@ -118,7 +118,7 @@ class RequestInfo {
   /**
    * @return the protocol of the request.
    */
-  virtual Http::Protocol protocol() const PURE;
+  virtual Optional<Http::Protocol> protocol() const PURE;
```

Comment:
Ultimately I think we will need this to be an enum specific to access logs if we don't split this apart later. On this topic, should this potentially print "TCP" ? (We could have it print "UDP" for example once we support UDP proxy). NBD, just a thought.

Reply:
Yeah, I think we'll want that eventually.  I wanted to limit the scope of this PR where I could.